gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,164 @@
+    id_seq = sqlalchemy.Sequence("batch_id_seq", metadata=Base.metadata, start=1000)
+
+    id = sqlalchemy.Column(
+        sqlalchemy.BigInteger,
+        nullable=False,
+        server_default=id_seq.next_value(),
+    )
+    name = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    project_id = sqlalchemy.Column(sqlalchemy.String(64), nullable=False)
+
+    members = orm.relationship("BatchMembership", back_populates="batch")
+
+    def __repr__(self):
+        created_datetime = (
+            self.created_datetime.isoformat() if self.created_datetime else None
+        )
+        updated_datetime = (
+            self.updated_datetime.isoformat() if self.updated_datetime else None
+        )
+
+        return "<Batch(id='{}', name='{}', project_id='{}', created_datetime='{}', updated_datetime='{}')>".format(
+            self.id, self.name, self.project_id, created_datetime, updated_datetime
+        )
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, self.__class__)
+            and self.id == other.id
+            and self.name == other.name
+            and self.project_id == other.project_id
+            and self.created_datetime == other.created_datetime
+            and self.updated_datetime == other.updated_datetime
+        )
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def to_dict(self):
+        created_datetime = (
+            self.created_datetime.isoformat() if self.created_datetime else None
+        )
+        updated_datetime = (
+            self.updated_datetime.isoformat() if self.updated_datetime else None
+        )
+
+        return {
+            "id": self.id,
+            "name": self.name,
+            "project_id": self.project_id,
+            "created_datetime": created_datetime,
+            "updated_datetime": updated_datetime,
+        }
+
+    def to_json(self):
+        """Returns a JSON safe representation of a batch"""
+        return json.loads(json.dumps(self.to_dict()))
+
+
+class BatchMembership(Base, audit.AuditColumnsMixin):
+    """Membership class to represent which nodes belong in a batch.
+
+    Attributes:
+        batch_id: id of the batch
+        node_id: id of the node
+        node_type: type of the node
+        created_datetime: the date and time when node was added to batch
+        updated_datetime: the date and time when node membership was last updated
+    """
+
+    __tablename__ = "batch_membership"
+    __table_args__ = (
+        schema.PrimaryKeyConstraint("batch_id", "node_id", name="batch_membership_pk"),
+        schema.ForeignKeyConstraint(
+            ("batch_id",),
+            ("batch.id",),
+            name="batch_membership_batch_id_fk",
+        ),
+    )
+
+    batch_id = sqlalchemy.Column(sqlalchemy.BigInteger, nullable=False)
+    node_id = sqlalchemy.Column(sqlalchemy.String(64), nullable=False)
+    node_type = sqlalchemy.Column(sqlalchemy.String(64), nullable=False)
```

Comment:
might as well change all of the `String` types to `Text`.  `node_id` is `Text` in the graph.  Program names, project codes, and node types are all unbounded in our system.